### DISCUSSION TITLE
Quick way to split a selection into start & end?

### DISCUSSION THREAD

**Discussion by lukasjuhrich** [helix-editor/helix#10722] [Q&A]
Hi,

I often see myself trying to change surrounding spaces to surrounding newlines, e.g. when wanting to change
```typst
Lorem ipsum $ x + y = z $ dolor sit amet
```
to
```typst
Lorem ipsum $
  x + y = z
$ dolor sit amet.
```
Currently, I use tree-sitter to select up until the inside of the $$ block, and then surround this block with a newline (`m s <ret>`). However, this does not trigger any alignment.

Is there an idiomatic way to split a given selection `$| x + y = z |$` into the start and end positions, so I can just `i <ret>` on both newly created selections and get the proper alignment?

**Comment by TornaxO7** [comment]
Hm... my first try would be to select the area with `ma$` and apply a `:select` regex, like `[(^$)(\$$)]`. But I'm a bit unsure as well.

**Reply by lukasjuhrich** [reply]
Hmmm… That regex and all the ones I could come up with yield a selection including dollar signs, which isn't particularly helpful: I don't have the possibility to navigate forward with the first while navigating backward with the second part of the selection.

I'd probably need lookahead and lookbehind for that, which `:select` doesn't appear to support.

**Reply by lukasjuhrich** [reply]
Ah, but there's a workaround! selecting `(\$ )( \$)` and then _subsequently selecting_ ` ` gives me the spaces I want.
However, for whatever reason I still don't get the proper indentation… maybe an LSP issue as well.

**Comment by kirawi** [comment]
This was sort of a regression introduced when we switched to regex-cursor. See https://github.com/helix-editor/helix/pull/9422#discussion_r1473022295
